controllers/ExerciseController.js

```diff
@@ -13,22 +13,58 @@ exports.getExoByLang = (req, res) => {
     })
 }
 
+/**
+ * Function used to get the exercice.
+ * @param {*} query 
+ * @param {*} req 
+ * @param {*} res 
+ */
+function showExercice(query, req, res){
+    Exercise.getExo(query,function(err, exercise){
+        let correctionText;
+        let skeletonText;
+
+        //Test if file correction exist
+        if(fs.existsSync(process.cwd() + `/corrections/${req.params.slug}.js`)){
+            correctionText = fs.readFileSync(process.cwd() + `/corrections/${req.params.slug}.js`, "utf-8");
+        }else{
+            correctionText = null;
+        }
+        
+        //Test if file skeletons exist
+        if(fs.existsSync(process.cwd() + `/skeletons/${req.params.slug}.js`)){
+            skeletonText = fs.readFileSync(process.cwd() + `/skeletons/${req.params.slug}.js`, "utf-8");
+        }else{
+            skeletonText = null;
+        }
+
+        //Return
+        //If correction + skeleton
+        if(correctionText != null && skeletonText != null){
+            res.render('ExerciseView', {exercise, menu:"exercises", correctionText, skeletonText});
+        }
+        //If only correction
+        if(correctionText != null && skeletonText == null){
+            res.render('ExerciseView', {exercise, menu:"exercises", correctionText});
+        }
+        //If only skeleton
+        if(correctionText == null && skeletonText != null){
+            res.render('ExerciseView', {exercise, menu:"exercises", skeletonText});
+        }
+        //If no one
+        if(correctionText == null && skeletonText == null){
+            res.render('ExerciseView', {exercise, menu:"exercises"});
```

Comment:
you can probably render only once and just pass parameters to the view whether or not it's null.
then you can just do your conditions in twig files